File: csharp/NemesisSystem/RelationshipEvent.cs
namespace NemesisSystem;

public class RelationshipEvent
{
    public string EventId { get; }
    public EventType EventType { get; }
    public Character Source { get; }
    public Character Target { get; }
    public string Description { get; }
    public DateTime Timestamp { get; }

    public RelationshipEvent(string eventId, EventType eventType, Character source, Character target, string description)
    {
        EventId = eventId;
        EventType = eventType;
        Source = source;
        Target = target;
        Description = description;
        Timestamp = DateTime.UtcNow;
    }
}

File: javascript/src/RelationshipEvent.ts
import { EventType } from './EventType';
import { Character } from './Character';

export class RelationshipEvent {
  eventId: string;
  eventType: EventType;
  source: Character;
  target: Character;
  description: string;
  timestamp: Date;

  constructor(eventType: EventType, source: Character, target: Character, description: string) {
    this.eventId = crypto.randomUUID();
    this.eventType = eventType;
    this.source = source;
    this.target = target;
    this.description = description;
    this.timestamp = new Date();
  }
}
